stdlib/builtin.rbi

```diff
@@ -564,7 +564,9 @@ class Encoding
 end
 
 class Regexp
-  def self.compile: (String, *any) -> Regexp
+  def self.compile: (String) -> Regexp
+                  | (String, bool) -> any
```

Comment:
The order of overloading matters.
`(String, bool) -> any` should be below (as far as I designed).

Reply:
Fixed via da3016a